xlators/protocol/client/src/client-rpc-fops.c

```diff
@@ -5100,6 +5111,13 @@ client3_3_lk(call_frame_t *frame, xlator_t *this, void *data)
             client_is_setlk(local->cmd)) {
             client_add_lock_for_recovery(local->fd, args->flock, &local->owner,
                                          local->cmd);
+        } else if (local->fix_open) {
+            xdata = dict_new();
+            if (xdata == NULL) {
+                op_errno = ENOMEM;
+                goto unwind;
+            }
+            set_strict_locks_in_dict(this, xdata);
```

Comment:
Line 2422 does this only when fix_open is set. Here it is done without any check. Which of the two is correct?

Reply:
Line 5114 checks this.